pennylane/devices/default_qubit.py

```diff
@@ -557,17 +557,17 @@ def preprocess_transforms(
 
         if config.interface == qml.math.Interface.JAX_JIT:
             transform_program.add_transform(no_counts)
-        transform_program.add_transform(
-            mid_circuit_measurements, device=self, mcm_config=config.mcm_config
-        )
-        # validate_device_wires needs to be after defer_measurement has added more wires.
-        transform_program.add_transform(validate_device_wires, self.wires, name=self.name)
         transform_program.add_transform(
             decompose,
             stopping_condition=stopping_condition,
             stopping_condition_shots=stopping_condition_shots,
             name=self.name,
         )
+        transform_program.add_transform(device_resolve_dynamic_wires, wires=self.wires)
+        transform_program.add_transform(
+            mid_circuit_measurements, device=self, mcm_config=config.mcm_config
+        )
+        transform_program.add_transform(validate_device_wires, self.wires, name=self.name)
```

Comment:
Is there anything that might happen in `mid_circuit_measurements` that might require us to run `decompose` a second time? 🤔